Add timezone support

eduardomozart opened the pull request:
## Problem

- Add timezone support for GLPI.
- Forum issue: https://forum.yunohost.org/t/showing-the-wrong-time-in-glpi-app/37837/5

## Solution

- *And how do you fix that problem*

## PR Status

- [X] Code finished and ready to be reviewed/tested
- [X] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

eduardomozart committed:
Update install

eduardomozart committed:
Update install

eduardomozart committed:
Update install

ericgaspar committed:
Merge branch 'testing' into pr/58

eduardomozart committed:
Update install

eduardomozart commented:
I've updated the command to MariaDB. Tested on YunoHost 13. Can you run the bot again?

ericgaspar commented:
!testme

DeMiro5001 commented:
Shouldn't be there a similar action in the upgrade script for older installed instances?

eduardomozart committed:
Implement timezone setup in upgrade script

Add timezone configuration setup in upgrade script.

eduardomozart committed:
Refactor timezone setup for database client detection

Updated timezone setup to use appropriate database client and tool based on availability.

eduardomozart committed:
Refactor timezone configuration script

Update timezone setup to use appropriate database tools based on availability.

DeMiro5001 commented:
I think it should check for the manifest version so this is only run for older versions. For my understanding, this should only be done once.
Can you try upgrading from the upgraded instance and see if it works ?

DeMiro5001 commented:
!testme

eduardomozart commented:
This command can be run more than once with no problems, but I agree it would be better to run it once. The only advantage to run it frequently is that if a new timezone is created it will be reflected in GLPI, but it’s not something that happens frequently.

eduardomozart committed:
Refactor timezone setup logic in upgrade script

eduardomozart committed:
Merge branch 'YunoHost-Apps:master' into patch-6

eduardomozart commented:
@DeMiro5001 added a version validation based on the manifest; it upgraded and ran the upgrade script just fine.

```
root@instance-20251122-1221:/root# yunohost app upgrade glpi --url https://github.com/eduardomozart/glpi_ynh/tree/patch-6
Info: Now upgrading glpi…
Info: Creating a safety backup prior to the upgrade
Info: Collecting files to be backed up for glpi…
Info: Declaring files to be backed up...
Info: Backing up the MySQL database...
Info: Backup script completed for glpi. (YunoHost will then actually copy those files to the archive).
Info: Creating a backup archive from the collected files…
Info: The archive will contain about 409MB of data.
Success! Backup created: glpi-pre-upgrade2
Info: Backup glpi-pre-upgrade1 was deleted because it is replaced by a newer backup glpi-pre-upgrade2
Info: Updating sources...
Info: Updating system_user...
Info: Updating install_dir...
Info: Updating permissions...
Info: Updating apt...
Info: Updating database...
Info: [+++.................] > Upgrading source files...
Info: [###+++..............] > Updating the database...
Info: [######++++..........] > --message=Configuring Fail2Ban...
Info: The service fail2ban has correctly executed the action reload-or-restart.
Info: [##########+++.......] > Upgrading system configurations related to glpi...
Info: [#############+++....] > Setting up time zones...
Info: [####################] > Upgrade of glpi completed
Success! glpi upgraded
```
